src/hooks/useMounted/useMounted.js

```diff
@@ -20,13 +20,6 @@ export const useMounted = () => {
     const mounted = useRef(false);
     useEffect(() => {
         mounted.current = true;
-        return () => mounted.current = false;
     }, []);
     return mounted.current;
-};
-
-export const useUnmounted = () => {
```

Comment:
Did you try to fix it with the solution I offered? (i.e. return a function the returns the value instead of the value itself)

Reply:
I believe the `useUnmounted` hook is not different from simply writing the logic yourself, because, the use case is different than `isMounted`, which can be useful in multiple places, but, when a developer wants to do something on `unmount` then it must be done within a *returned function* within a `useEffect` (for example clearing *timeouts*).

So my point is, that this specific `useUnmounted` hook, as it works now, as a flag (function which returns a boolean), is not very helpful, because once a component has been unmounted, only the returned functions inside the `useEffects` executes, so it's not clear in which scenario this hook will save you writing more code..

Normally, everything you want to "undo" when a component unmounts are things you write inside a `useEffect`, like events listeners and timeouts, so most likely a developer already has a `useEffect` written in their component, and all is needed to be added is the returned function, which is not a lot to add

```js
const Foo = () => {
  const isUnmounted = useUnmounted()
   
  useEffect(() => {
	  () => {
	     // this will be the only scope which will run once un-mounted,
	     // so any event listener removal or clearing timeouts is done here
	  }
  }, [])
  
  console.log( isUnmounted() ) // will no reach this line of code if the component has been un-mounted

  return null;
}
```

Reply:
Right, that totally makes sense. I would just check the current usage of this hook in C1 to make sure we're not missing some cases where it could be useful. Because I don't think we just made this hook for completeness' sake.

Reply:
@gaeaehrlich - can you check again all the places it is being used, just to be certain there's no special scenario where this hook was used where a simple useEffect with a return function wouldn't do

Reply:
@yairEO All use cases seem standard (async state update)